Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -43,27 +44,53 @@ public FreezeOnMatchCustomization(Type targetType)
         /// that will be satisfied by the frozen specimen.
         /// </param>
         /// <exception cref="ArgumentNullException">
-        /// <paramref name="targetType"/> or<paramref name="matcher"/> is null.
+        /// <paramref name="targetType"/> or <paramref name="matcher"/> is null.
         /// </exception>
         public FreezeOnMatchCustomization(
             Type targetType,
             IRequestSpecification matcher)
+            : this(new TypeElement(targetType), matcher)
         {
-            if (targetType == null)
+            this.TargetType = targetType;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FreezeOnMatchCustomization"/> class.
+        /// </summary>
+        /// <param name="reflectionElement">
+        /// The <see cref="IReflectionElement"/> used to create specimens to freeze.
+        /// </param>
+        /// <param name="matcher">
+        /// The <see cref="IRequestSpecification"/> used to match the requests
+        /// that will be satisfied by the frozen specimen.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="reflectionElement"/> or <paramref name="matcher"/> is null.
+        /// </exception>
+        public FreezeOnMatchCustomization(
+            IReflectionElement reflectionElement,
+            IRequestSpecification matcher)
+        {
+            if (reflectionElement == null)
             {
-                throw new ArgumentNullException(nameof(targetType));
+                throw new ArgumentNullException(nameof(reflectionElement));
             }
 
             if (matcher == null)
             {
                 throw new ArgumentNullException(nameof(matcher));
             }
 
-            this.TargetType = targetType;
+            this.ReflectionElement = reflectionElement;
             this.Matcher = matcher;
         }
 
         /// <summary>
+        /// The <see cref="ReflectionElement"/> describing the specimen to freeze.
+        /// </summary>
+        public IReflectionElement ReflectionElement { get; }
+
+        /// <summary>
         /// The <see cref="Type"/> of the frozen specimen.
         /// </summary>
         public Type TargetType { get; }
```

Comment:
This PR changes some _core_ classes ([FreezeOnMatchCustomization](https://github.com/sergeyshushlyapin/AutoFixture/blob/13513fb9507b3d33f56dc69d2dbd21d6727c48a0/Src/AutoFixture/FreezeOnMatchCustomization.cs)) as well as Glue Library classes ([xUnit2.FrozenAttribute](https://github.com/sergeyshushlyapin/AutoFixture/blob/13513fb9507b3d33f56dc69d2dbd21d6727c48a0/Src/AutoFixture.xUnit.net2/FrozenAttribute.cs)). I started it as xUnit.net2 contribution and intended to fix all the other libraries with follow up requests.

If I deprecate the `TargetType` property and [corresponding](https://github.com/sergeyshushlyapin/AutoFixture/blob/13513fb9507b3d33f56dc69d2dbd21d6727c48a0/Src/AutoFixture/FreezeOnMatchCustomization.cs#L33) [constructors](https://github.com/sergeyshushlyapin/AutoFixture/blob/13513fb9507b3d33f56dc69d2dbd21d6727c48a0/Src/AutoFixture/FreezeOnMatchCustomization.cs#L49) (which are in 'core'), then other libraries start failing:

```
  1) Building C:\sc\autofixture\Src\AutoFixture.NUnit2.sln failed with exitcode 1.
  2) CS0618: FrozenAttribute.cs(113,20): 'FreezeOnMatchCustomization.FreezeOnMatchCustomization(Type, IRequestSpecification)' is obsolete: 'This constructor has been deprecated. Please use constructor that takes IReflectionElement parameter.'
```

What is the best approach to go? Should I suppress the error in other libraries for the time being or postpone obsoleting the property and constructors?